packages/x-date-pickers/src/DateCalendar/tests/DateCalendar.spec.tsx

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { DateCalendar } from '@mui/x-date-pickers/DateCalendar';
+
+<DateCalendar />;
+
+<DateCalendar defaultValue={new Date()} />;
+
+<DateCalendar value={new Date()} />;
```

Comment:
The core does not do it, I think because it brings to much complexity to the typing.

Note that it's totally fine to pass none of those props, `<DateCalendar />` is a valid usage.

Reply:
Gotcha. Well—it's probably safer, but I was just trying to think about DX—wouldn't it be best to error out if component is missing the required props to function correctly. 🤔 
On the other hand—maybe I'm just looking through too narrow windows and users have cases where they don't need to bind value, but just care about listening to `onChange` and reacting to it. 👌

Reply:
For me the following make sense

```tsx
// Example: Whenever a date is picked, I close the picker then go on a new page (i.e: a train ticket website)
<DateCalendar onChange={handleDateChange} />

// Same but with a default value
<DateCalendar defaultValue={new Date()} onChange={handleDateChange} /> 

// Classic controlled mode
<DateCalendar value={new Date()} onChange={handleDateChange} /> 
```

I agree that use cases without `onChange` are pretty strange but I might be missing some.
I'd favor following the core behavior on that one, just to be consistent. I don't think people forget to pass those props.

Reply:
Exactly—those are perfectly viable use-cases. 👌